src/test/java/com/google/sps/GetPlayerAccessoriesTest.java

```diff
@@ -0,0 +1,205 @@
+    this.getPlayerAccessories = new GetPlayerAccessories();
+    this.getPlayerAccessories.init();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /** doGet method should respond with Json message with all accessories available to the player */
+  @Test
+  public void testThatDoGetRespondsWithAllPlayerAccessories() throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+
+    // add user and all their available accessories
+    User user = this.localUserService.getCurrentUser();
+    List<String> playerAccessoryIDs = Arrays.asList(HAT_ID, GLASSES_ID, COMPANION_ID);
+    createCurrentPlayerAndAddToDatabase(user, playerAccessoryIDs);
+
+    String accessory1Id = HAT_ID;
+    String accessory2Id = GLASSES_ID;
+    String accessory3Id = COMPANION_ID;
+    Accessory accessory1 = createAccessoryAndAddToDatabase(accessory1Id);
+    Accessory accessory2 = createAccessoryAndAddToDatabase(accessory2Id);
+    Accessory accessory3 = createAccessoryAndAddToDatabase(accessory3Id);
+
+    String expectedJson = gson.toJson(Arrays.asList(accessory1, accessory2, accessory3));
+
+    this.getPlayerAccessories.doGet(this.request, this.response);
+
+    // checks that getPlayerAccessories responds with a Json string containing each accessory
+    // object
+    String expected = expectedJson;
+    String result = stringWriter.toString();
+    Assert.assertTrue(result.contains(expected));
+  }
+
+  /** when an accessory is not found, it is not included in response and gameplay continues */
+  @Test
+  public void testThatNotFoundAccessoriesAreNotIncludedInResponse() throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(this.response.getWriter()).thenReturn(printWriter);
+
+    // add user and all their available accessories
+    User user = this.localUserService.getCurrentUser();
+    List<String> playerAccessoryIDs = Arrays.asList(HAT_ID, GLASSES_ID, COMPANION_ID);
+    createCurrentPlayerAndAddToDatabase(user, playerAccessoryIDs);
+
+    // the companion id is not added to database, so a EntityNotFoundException will be thrown
+    String accessory1Id = HAT_ID;
+    String accessory2Id = GLASSES_ID;
+    Accessory accessory1 = createAccessoryAndAddToDatabase(accessory1Id);
+    Accessory accessory2 = createAccessoryAndAddToDatabase(accessory2Id);
+
+    String expectedJson = gson.toJson(Arrays.asList(accessory1, accessory2));
+
+    this.getPlayerAccessories.doGet(this.request, this.response);
+
+    // checks that getPlayerAccessories responds with a Json string containing each accessory
+    // object, but without the accessory not in the database
+    String expected = expectedJson;
+    String result = stringWriter.toString();
+    Assert.assertTrue(result.contains(expected));
```

Comment:
I get that this will confirm that the added accessories are found int the result, but to be more accurate to the test name, maybe assert true that no other items are in the result instead of the expected items being found in the result

Reply:
You makes a great point. Changed to comparing for exact equality with JsonElements (issue is that StringWriter toString() is not exactly the same as the JSON).